components/welcome_post_renderer/index.test.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {mount} from 'enzyme';
+
+import {Provider} from 'react-redux';
+
+import WelcomePostRenderer from 'components/welcome_post_renderer';
+import {GlobalState} from 'types/store';
+import {mockStore} from 'tests/test_store';
+import {Post} from '@mattermost/types/posts';
+
+describe('components/WelcomePostRenderer', () => {
+    const initialStore = {
+        entities: {
+            channels: {
+                currentChannelId: 'current_channel_id',
+                channels: {
+                    current_channel_id: {id: 'current_channel_id'},
+                },
+            },
+            users: {
+                currentUserId: 'user',
+                profiles: {
+                    user: {id: 'user', roles: 'system_user'},
+                    admin: {id: 'admin', roles: 'system_admin'},
+                },
+            },
+            general: {
+                config: {},
+            },
+            teams: {teams: {current_team_id: {}}, currentTeamId: 'current_team_id'},
+            posts: {posts: {}},
+            groups: {
+                groups: {},
+                myGroups: [],
+            },
+            preferences: {myPreferences: {}},
+            emojis: {},
```

Comment:
I notice that in your mock state have several empty objects. Do the state of a particular test require those? if not, it is better to remove them, cause this cause polution in the code.

Reply:
Yeah I added them one by one until it worked 😅 Because I'm using the Markdown parser it requires quite a lot of stuff from redux (emojis, custom groups, etc..)